test/frontend/reducers/test_concept_systems.js

```diff
@@ -1,12 +1,14 @@
 import { expect } from '../test_helper';
-import conceptSystems  from '../../../webpack/reducers/concept_systems_reducer';
-import _ from 'lodash';
+
+import { byIdReducer } from '../../../webpack/reducers/reducer_generator';
```

Comment:
I needed to write reducers for systems and programs. I realized that they would basically be the same except for the action type that was passed in. I decided to write some functions that can generate reducers. I also got rid of a bunch of reducers. Hopefully this will cut down on boilerplate in the application.